Add action to take time step on FD grid

nilsdeppe opened the pull request:
## Proposed changes

Exactly as the name says.

The action basically farms out to a system-specific implementation since if we've fallen back to FD the generic solvers just aren't robust enough. 

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

nilsdeppe committed:
Add action to take time step on FD grid

fmahebert commented on src/Evolution/DgSubcell/Actions/TakeTimeStep.hpp:
```diff
@@ -0,0 +1,104 @@
+/// \endcond
+
+/// \ingroup DgSubcellGroup
+/// \brief %Actions specific to using a finite-difference subcell method.
```
Can the namespace documentation be moved to a more generic header?

fmahebert commented on tests/Unit/Evolution/DgSubcell/Actions/Test_TakeTimeStep.cpp:
```diff
@@ -0,0 +1,215 @@
+struct Metavariables {
+  static constexpr size_t volume_dim = Dim;
+  using component_list = tmpl::list<component<Dim, Metavariables>>;
+  // using system = System<Dim>;
```
Delete

nilsdeppe committed:
Document evolution::dg::subcell::fd::Actions namespace

nilsdeppe commented:
I've:
- rebased
- squashed in the removal of the commented out line
- prepended a commit that documents the namespace in `DgSubcell/Actions/Actions.hpp`

Thanks for the review :)

kidder commented on src/Evolution/DgSubcell/Actions/TakeTimeStep.hpp:
```diff
@@ -0,0 +1,102 @@
+ * second argument.
+ *
+ * GlobalCache: nothing
+ * DataBox:
```
blank line to render correctly

kidder reviewed (changes requested):
you can squash immediately

nilsdeppe commented:
I've rebased and squashed. Thanks for the reviews :)
